routing/RouteSolver.py

```diff
@@ -0,0 +1,134 @@
+##                from the users start node to the users end-node. This data    ##
+##                will be piped to a Raspberry Pi that will then send the       ##
+##                data to an arduino for roaming the map.                       ##
+##################################################################################
+import Graph
+import sys
+import re
+##################################################################################
+##                                                                              ##
+##                            Global Variables                                  ##
+##                                                                              ##
+##################################################################################
+graph = Graph.Graph()
+##################################################################################
+##                                                                              ##
+##                            Functions/Methods                                 ##
+##                                                                              ##
+##################################################################################
+
+def is_int(s):
+    try:
+        int(s)
+        return True
+    except ValueError:
+        return False
+
+def loadConfig(configFile):
+    f = open(configFile,"r")
+    lines = f.readlines()
+    currNode = ""
+    bad_chars = '(){}<>[]'
+    for i in range(0,len(lines)):
+        line= lines[i]
+        if "Next Node:" in line:
+            graph.add_vertex(lines[i+1].strip())
+            currNode = lines[i+1].strip()
+        elif "Connected Nodes:" in line:
+            connectedNodes = lines[i+1].split(';')
+            for node in connectedNodes:
+                nodeName = node.split(',')
+                for c in bad_chars: nodeName[0] = nodeName[0].replace(c, "")
+                for c in bad_chars: nodeName[1] = nodeName[1].replace(c, "")
+                nodeName[1].rstrip()
+                #graph.add_edge((currNode,(nodeName[0],nodeName[1])))
+                # Adding directions is causing an issue. May have to come up with
+                #another solution for direction
+                graph.add_edge((currNode,nodeName[0]))
+
+def getGraphDetailsFromUser():
+    numberOfNodes = 0;
+    while True:
+        numberOfNodes = input("How many nodes are in your graph? ")
+        if(is_int(numberOfNodes)):
+            break
+    nodeCount = 0
+    while (nodeCount < int(numberOfNodes)):
+        newName = input("Please enter a name for node {}\n".format(nodeCount))
+        graph.add_vertex(newName)
+        nodeCount += 1
+    print(graph.vertices())
+    for node in graph.vertices():
+        anotherNode = True
+        while anotherNode:
+            nodeName = input("Please enter connected node name for node {}:\n".format(node))
+            if(nodeName in graph.vertices()):
```

Comment:
Parenthesis around if statements are unnecessary. `if node_name in graph.vertices():`